conftest.py

```diff
@@ -441,3 +477,24 @@ def Npart(request):
     Needed because indirect=True is used for loading the datasets.
     """
     return request.param
+
+
+@pytest.fixture(scope="session")
+def dask_client_fixture(request):
+    """
+    Fixture for spinning up a dask LocalCluster and Client
+    """
+
+    # there are a number of options, including using some of the internal dask
+    # functions for testing, but this seems the simplest way to ensure only a
+    # single cluster is spun up during testing. Some useful discussion here:
+    # https://github.com/dask/distributed/issues/4479#issuecomment-772119435
+    # also note that if running pytest in parallel, this fixture will need
```

Comment:
just curious, *do* we ever run pytest in parallel ?

Reply:
not that I know of! but `setup.cfg` does have `pytest-xdist~=2.1.0` in the test requirements. So I suspect someone was experimenting with it at some point...

Reply:
oh yeah, good catch. Well I guess we'd better _remove_ this from the requirements then. We have enough on our plate with the pytest migration, making sure it also works in parallel seems to be very low priority. Doesn't have to be in this PR, though I wouldn't be against it